pulp_container/app/authorization.py

```diff
@@ -156,45 +159,51 @@ def has_pull_permissions(self, path):
             distribution = ContainerDistribution.objects.get(base_path=path)
         except ContainerDistribution.DoesNotExist:
             return False
-        if not distribution.private:
-            return True
-        if self.user.has_perm("container.pull_containerdistribution", distribution):
-            return True
-        if self.user.has_perm("container.pull_containerdistribution"):
-            return True
 
-        return False
+        # Fake the pull request
+        request = Request(HttpRequest())
+        request.method = "GET"
+        request.user = self.user
+        request._full_data = {"base_path": path}
+        # Fake the corresponding view
+        view = namedtuple("FakeView", ["action", "get_object"])("pull", lambda: distribution)
+        return self.access_policy.has_permission(request, view)
 
     def has_push_permissions(self, path):
         """
         Check if the user has permissions to push to the repository specified by the path.
         """
-
         try:
             distribution = ContainerDistribution.objects.get(base_path=path)
         except ContainerDistribution.DoesNotExist:
-            # distribution does not exist, need to create a new one
-            if not self.user.has_perm("container.add_containerdistribution"):
-                return False
-            namespace = path.split("/")[0]
-            return NamespacePermissionsChecker.has_permissions(
-                namespace, self.user, "container.manage_namespace_distributions"
-            )
-        # this is an existing distribution. User should have model or object perms
-        if self.user.has_perm("container.push_containerdistribution") or self.user.has_perm(
-            "container.push_containerdistribution", distribution
-        ):
-            return True
-        return False
+            distribution = None
+
+        # Fake the pull request
```

Comment:
s/pull/push/

Reply:
Let's only "Fake the request" then we don't introduce more copy and paste errors.